### PR TITLE
[FE-16381] Muted Colorization and Missing Lines after applying Chart Filter

### PR DESCRIPTION
**Pull request by cjvil**
See slack discussion: https://heavy-ai.slack.com/archives/G011G8MFZAP/p1676309658535929?thread_ts=1675986436.157549&cid=G011G8MFZAP

Count by group was being used for sampling calculations, so applied sample ratios did not take into account chart level filters. Adds a lastFilteredSize method to raster layers (in general, as this would affect all sampled charts) as per-layer crossfilter does include chart filters, and adds a preflight query to get size per layer before generating vega. Group counts are used for other charts, so there are some mixins and points in the code where lastFilteredSize for the group is still used (e.g. count charts, non-sampled charts). Contour doesn’t seem to apply sampling, at least along the altered code path, needed one exception but otherwise seems to have the correct combo of mixins to work fine. 

# Merge Checklist
## :wrench: Issue(s) fixed:
- [ ] Author referenced issue(s) fixed by this PR:
- [ ] Fixes #0

## :smoking: Smoke Test
- [ ] Works in chrome
- [ ] Works in firefox
- [ ] Works in safari
- [ ] Works in ie edge
- [ ] Works in ie 11

## :ship: Merge
- [ ] author crafted PR's title into release-worthy commit message.

### CODE REVIEW

File: src/charts/raster-chart.js [hunk 1]
```diff
@@ -491,11 +504,7 @@ export default function rasterChart(parent, useMap, chartGroup, _mapboxgl) {
     const bounds = _chart.getDataRenderBounds()
     _chart._updateXAndYScales(bounds)
 
-    _vegaSpec = genLayeredVega(
-      _chart,
-      group,
-      lastFilteredSize(group.getCrossfilterId())
-    )
+    _vegaSpec = genLayeredVega(_chart)
```
Comment (cjvil) [inline src/charts/raster-chart.js:507]: Seemed to be using an old function signature; `genLayeredVega` currently only takes one `chart` parameter

**Comment by mapd-bot-os** [comment]
build succeeded

**Comment by mapd-bot-os** [comment]
build succeeded

**Comment by mapd-bot-os** [comment]
build succeeded

**Review by johallar — APPROVED** [review]

**Pull request merged**